File: Problems/Valid Parentheses/ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        
        P = []

        for c in s:
            if (c == '(' or c == '[' or c == '{'):
                P.append(c)
            elif P == []:
                return False
            elif (c == ')'):
                if (P.pop() != '('):
                    return False
            elif (c == ']'):
                if (P.pop() != '['):
                    return False
            elif (c == '}'):
                if (P.pop() != '{'):
                    return False
            else:
                return False
        
        return P == []

File: Problems/Valid Parentheses/ValidParentheses.c
#include <string.h>

bool isValid(char* s) {

    char P[strlen(s)];
    int end = -1;

    for (int i = 0; s[i] != '\0'; i++) {

        if (s[i] == '(' || s[i] == '[' || s[i] == '{') {
            P[++end] = s[i];
        }
        else if (end == -1) {
            return false;
        }
        else if (s[i] == ')') {
            if (P[end--] != '(') {
                return false;
            }
        }
        else if (s[i] == ']') {
            if (P[end--] != '[') {
                return false;
            }
        }
        else if (s[i] == '}') {
            if (P[end--] != '{') {
                return false;
            }
        }

    }

    return end == -1;
    
}
